plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestCachingDirectoryLister.java

```diff
@@ -0,0 +1,282 @@
+                    binder.bind(DirectoryLister.class).toInstance(cachingDirectoryLister);
+                    binder.bind(DirectoryCallback.class).toInstance(cachingDirectoryLister);
+                })
+                .build();
+    }
+
+    @Test
+    public void testCacheInvalidationIsAppliedSpecificallyOnTheTableBeingChanged()
+    {
+        String table1Name = "cached_table1_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + table1Name + " (col1 int) WITH (format = 'ORC')");
+        assertUpdate("INSERT INTO " + table1Name + " VALUES (1), (2), (3)", 3);
+        // The listing for the table1 should be in the directory cache after this call
+        assertQuery("SELECT sum(col1) FROM " + table1Name, "VALUES (6)");
+        org.apache.hadoop.fs.Path table1Location = getTableLocation(TPCH_SCHEMA, table1Name);
+        assertTrue(cachingDirectoryLister.isCached(table1Location));
+
+        String table2Name = "cached_table2_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + table2Name + " (col2 int) WITH (format = 'ORC')");
+        assertUpdate("INSERT INTO " + table2Name + " VALUES (11), (12)", 2);
+        // The listing for the table2 should be in the directory cache after this call
+        assertQuery("SELECT sum(col2) FROM " + table2Name, "VALUES (23)");
+        org.apache.hadoop.fs.Path table2Location = getTableLocation(TPCH_SCHEMA, table2Name);
+        assertTrue(cachingDirectoryLister.isCached(table2Location));
+
+        assertUpdate("INSERT INTO " + table1Name + " VALUES (4), (5)", 2);
+        // Inserting into the table1 should invalidate the cached listing of the files belonging only to the table1.
+        assertFalse(cachingDirectoryLister.isCached(table1Location));
+        assertTrue(cachingDirectoryLister.isCached(table2Location));
+
+        assertQuery("SELECT sum(col1) FROM " + table1Name, "VALUES (15)");
+        assertQuery("SELECT sum(col2) FROM " + table2Name, "VALUES (23)");
+
+        assertUpdate("DROP TABLE " + table1Name);
+        assertUpdate("DROP TABLE " + table2Name);
+    }
+
+    @Test
+    public void testInsertIntoNonPartitionedTable()
+    {
+        String tableName = "insert_into_non_partitioned_table_" + randomTableSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (col1 int) WITH (format = 'ORC')");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1), (2), (3)", 3);
+        // The listing for the table should be in the directory cache after this call
+        assertQuery("SELECT sum(col1) FROM " + tableName, "VALUES (6)");
+        assertTrue(cachingDirectoryLister.isCached(getTableLocation(TPCH_SCHEMA, tableName)));
+        assertUpdate("INSERT INTO " + tableName + " VALUES (4), (5)", 2);
+        // Inserting into the table should invalidate the cached listing of the files belonging to the table.
+        assertFalse(cachingDirectoryLister.isCached(getTableLocation(TPCH_SCHEMA, tableName)));
```

Comment:
should this fail, assertTrue(exp) or assertFalse(exp) produces an unfriendly message.
It's not know what failed unless you check the stacktrace and match the lines with the code.

add a message

Reply:
I switched to `assertThat` to be consistent with the other tests.